### ISSUE TITLE
Add counts to Range Components (where appropriate)

### ISSUE THREAD

**Issue by davidklebanoff** [appbaseio/reactivesearch#269]
**Issue Type:**
enhancement

**Description:**
Some of the Range Components such as `SingleRange`, `MultiRange`, `SingleDropdownRange`, `MultiDropdownRange`, and `RatingsFilter` should support counts like the List Components.

**Comment by metagrover** [comment]
Components like `SingleRange`, `MultiRange`, `SingleDropdownRange` and `MultiDropdownRange` take the label (and value) from the user. Hence we do not run any sort of aggregations query there, and therefore no count data can be shown there.

**Comment by davidklebanoff** [comment]
But, unless I'm missing something, there's no reason you couldn't run an aggregations query, user-submitted labels or not?

**Comment by metagrover** [comment]
We certainly can, but need more thought on this. I think we can make it available via a boolean prop, which when true will run the aggs query and render count values.

Thoughts @siddharthlatest?

**Comment by davidklebanoff** [comment]
Related to #270.

**Comment by siddharthlatest** [comment]
I can see this as useful in some cases. We can use the `showCount` prop (defaults to `false`) for *Range and *DataList components to run a terms aggs query on the values and show the count.

**Comment by davidklebanoff** [comment]
@siddharthlatest - exactly

I think that UI quirk is acceptable and expected. To me personally, I think it's preferable, because I find the lag time waiting on the results and counts in the current non-data list components can sometimes be frustrating as it prevents people from quickly seeing the results (they are currently delayed by loading counts, which perhaps should be its own issue) and it prevents people from quickly adding multiple filters (they must click, then wait on counts, then click, then wait on counts, etc.). This approach of instant label loading and side-loaded counts is a best of both worlds compromise to me.

And of course this makes rooms for an additional configuration option, if you so choose. Developers could choose whether they want the counts to show up after the labels have loaded, or if they want the labels to show at the same time as counts.

It may also make sense to offer the option of whether label with counts that end up coming back 0 should be hidden or shown as a 0. If a developer picks the option where counts load in after labels, it might be weird if labels suddenly disappear (b/c they had 0 counts) a few moments after they filtered.

**Comment by siddharthlatest** [comment]
We can use the [`Histogram` aggregation](https://www.elastic.co/guide/en/elasticsearch/reference/current/search-aggregations-bucket-histogram-aggregation.html) to do this and a prop like `showCount`, which defaults to `false` to configure if counts should be shown.
